lib/wisper/sidekiq.rb

```diff
@@ -28,14 +28,34 @@ def self.register
 
     def broadcast(subscriber, publisher, event, args)
       options = sidekiq_options(subscriber)
-      Worker.set(options).perform_async(::YAML.dump([subscriber, event, args]))
+      schedule_options = sidekiq_schedule_options(subscriber, event)
+
+      if delay_value = schedule_options[:delay]
+        Worker.set(options).perform_in(delay_value, ::YAML.dump([subscriber, event, args]))
+      else
+        Worker.set(options).perform_async(::YAML.dump([subscriber, event, args]))
+      end
     end
 
     private
 
     def sidekiq_options(subscriber)
       subscriber.respond_to?(:sidekiq_options) ? subscriber.sidekiq_options : {}
     end
+
+    def sidekiq_schedule_options(subscriber, event)
+      return { } unless subscriber.respond_to?(:sidekiq_schedule_options)
+      return { } unless (options = subscriber.sidekiq_schedule_options).is_a?(Hash)
+
+      delay_option(options).merge( delay_option(options[event.to_sym]) )
```

Comment:
Definetly! I like what the last one that you propose, because it's easy to follow what is going on!

Reply:
Yeah, it took me a moment to figure out what that line of code was doing and why. I just thought it might be better spelt out for future readers of the code.